src/msg/msg_types.h

```diff
@@ -393,7 +416,7 @@ struct entity_addr_t {
       return false;
     if (is_blank_ip() || o.is_blank_ip())
       return true;
-    if (memcmp(&u, &o.u, sizeof(u)) == 0)
+    if (is_same_host(o))
```

Comment:
i think, after your change `memcpy()` should serve the purpose just well?

Reply:
@tchaikov 
Until things change again, especially with we compare structs where it could be unclear what is actually stored after the sa_address/sa_data field. Even more so when we start comparing structs that originate from Linux and FreeBSD on a hybrid cluster. 
So for most purposes I would like te get ride of all memcmp's here, but the `<, >, => =<` ones are a bit harder to get the logic right.

Reply:
on the contrary, i prefer `memcmp()` over the hand-written `<`. because every time we send a message, we look up the connection by the peer address. so it's sitting in the critical path, and we need to make it as fast as possible.